client/state/data-layer/wpcom/comments/index.js

```diff
@@ -160,35 +159,66 @@ export const announceFailure = ( { dispatch, getState }, { siteId, postId } ) =>
 	dispatch( errorNotice( error ) );
 };
 
-export const deleteComment = ( { dispatch }, action ) => {
-	const { siteId, commentId } = action;
+// @see https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/comments/%24comment_ID/delete/
+export const deleteComment = ( { dispatch, getState }, action ) => {
+	const { siteId, postId, commentId } = action;
+	const comment = getSiteComment( getState(), siteId, commentId );
+
+	dispatch( removeComment( siteId, postId, commentId ) );
+
 	dispatch(
-		http( {
-			method: 'POST',
-			apiVersion: '1.1',
-			path: `/sites/${ siteId }/comments/${ commentId }/delete`,
-			onSuccess: {
-				...action,
-				type: COMMENTS_DELETE_SUCCESS,
+		http(
+			{
+				method: 'POST',
+				apiVersion: '1.1',
+				path: `/sites/${ siteId }/comments/${ commentId }/delete`,
 			},
-			onFailure: {
+			{
 				...action,
-				type: COMMENTS_DELETE_FAILURE,
-			},
-		} )
+				comment,
+			}
+		)
 	);
 };
 
-export const removeCommentFromState = ( { dispatch }, action ) => {
-	const { siteId, postId, commentId } = action;
-	dispatch( removeComment( siteId, postId, commentId ) );
+export const announceDeleteSuccess = ( { dispatch } ) => {
+	dispatch(
+		createNotice(
+			'is-error',
+			translate( 'Comment deleted permanently.' ),
+			{
+				duration: 5000,
+				isPersistent: true,
+			}
+		)
+	);
```

Comment:
I personally like having some feedback when I perform actions. @drw158 what do you think?